crates/router/src/connector/payme/transformers.rs

```diff
@@ -1,109 +1,331 @@
-use masking::Secret;
+use api_models::payments::PaymentMethodData;
+use common_utils::pii;
+use error_stack::{IntoReport, ResultExt};
+use masking::{ExposeInterface, Secret};
 use serde::{Deserialize, Serialize};
 
 use crate::{
-    connector::utils::PaymentsAuthorizeRequestData,
+    connector::utils::{
+        missing_field_err, AddressDetailsData, CardData, PaymentsAuthorizeRequestData, RouterData,
+    },
     core::errors,
-    types::{self, api, storage::enums},
+    types::{self, api, storage::enums, MandateReference},
 };
 
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Serialize, Eq, PartialEq)]
-pub struct PaymePaymentsRequest {
-    amount: i64,
+#[derive(Debug, Serialize)]
+pub struct PayRequest {
+    buyer_name: Secret<String>,
+    buyer_email: pii::Email,
+    payme_sale_id: String,
+    #[serde(flatten)]
     card: PaymeCard,
 }
 
-#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[derive(Debug, Serialize)]
+pub struct MandateRequest {
+    currency: enums::Currency,
+    sale_price: i64,
+    transaction_id: String,
+    product_name: String,
+    sale_return_url: String,
+    seller_payme_id: Secret<String>,
+    sale_callback_url: String,
+    buyer_key: String,
```

Comment:
This field is `Secret` elsewhere, but is not `Secret` here. Was this missed out, or does this not need to be a `Secret`?